### ISSUE TITLE
Python crashes when trying to compute the gradient of jnp.take

### ISSUE THREAD

**Issue by bitsandscraps** [jax-ml/jax#25284]
### Description

When running the following script:
```python
from jax import grad, numpy as jnp

def loss_fn(W):
    return jnp.mean(jnp.take(W, 0, axis=-1))

grads = grad(loss_fn)(jnp.arange(3.0).reshape(3, 1))
```
Python crashes with the following error:
```bash
[1]    85044 IOT instruction (core dumped)  poetry run python
```
I ran into this error while trying to use `optax`'s `softmax_cross_entropy_with_integer_labels` and found that `jnp.take` on this [line](https://github.com/google-deepmind/optax/blob/3f0a64bdb7a3b5c03c09424e2ee48888fdd6173e/optax/losses/_classification.py#L341) was causing the error.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.4.35
jaxlib: 0.4.34
numpy:  1.26.4
python: 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0]
device info: NVIDIA GeForce RTX 4090-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='sisrel-jongeui-machine', release='6.8.0-49-generic', version='#49-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov  4 02:06:24 UTC 2024', machine='x86_64')                                                                                                                                                                                                                                                                                                                                                              $ nvidia-smi
Thu Dec  5 21:18:28 2024                                                                                                              +-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.127.05             Driver Version: 550.127.05     CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4090        Off |   00000000:01:00.0 Off |                  Off |
|  0%   40C    P8             10W /  450W |     401MiB /  24564MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A     85255      C   .../wisrl/offline-jax/.venv/bin/python        394MiB |
+-----------------------------------------------------------------------------------------+
```

**Comment by jakevdp** [comment]
I'm unable to reproduce this locally or on Colab, in either a CPU or GPU (T4) runtime.

It's possibly related to the particular GPU hardware you're using, but I'm not sure how to diagnose that.

Does this reproduce for other jax/jaxlib versions?

**Comment by bitsandscraps** [comment]
The bug disappeared when I downgraded `jax` to `0.4.34`.  By the way, I forgot to mention this warning message before the core dumped line.
```bash
2024-12-06 03:05:35.451686: F external/xla/xla/service/scatter_determinism_expander.cc:70] Check failed: scatter_shape.dimensions_size() == 1 (2 vs. 1)
```

**Comment by hawkinsp** [comment]
Can you try on the newly released jax 0.4.36?

**Comment by bitsandscraps** [comment]
It works now. Thank you.
